### PR TITLE
aamath: Fix build for Linux

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [x] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

Gist logs: https://gist.github.com/f35fbd08bfda962a28be9668dcc5a340

### CODE REVIEW

**Comment by sjackman** [comment]
```
g++ -Wall -O2 -g -DUSE_READLINE -c aamath.cc
722
aamath.cc:19:31: fatal error: readline/readline.h: No such file or directory
723
```
https://github.com/Homebrew/linuxbrew-core/pull/19520/checks?check_run_id=430766528#step:5:722

File: Formula/aamath.rb [hunk 1]
```diff
@@ -15,6 +15,11 @@ class Aamath < Formula
     sha256 "0212e0b5844ea1a491bc7d4fcab2b590921042b28bc50e79c36cd9e15d08e2aa" => :mavericks
   end
 
+  unless OS.mac?
+    depends_on "bison" => :build # for yacc
+    depends_on "flex" => :build
```
Comment (sjackman) [inline Formula/aamath.rb]:
```suggestion
    uses_from_macos "flex" => :build
```
Does `uses_from_macos "…" => :build` work?

Comment (issyl0) [reply Formula/aamath.rb]: It does! I'm never sure which dependencies are macOS-provided until I have time to search Homebrew/homebrew-core for the `uses_from_macos` equivalent.

**Comment by issyl0** [comment]
```
==> brew linkage --test aamath
Missing libraries:
  libreadline.so.8
==> FAILED
```

What more does it want from me?!

**Pull request merged**